From 16127c54bd8664b84a1a55aca45507e80f02695c Mon Sep 17 00:00:00 2001
From: Alexey Makhalov <amakhalov@vmware.com>
Date: Thu, 1 Apr 2021 03:28:10 +0000
Subject: [PATCH] cgroup v1: cgroup.stat support

Enabled cgroup statistic for cgroups version 1.

Signed-off-by: Alexey Makhalov <amakhalov@vmware.com>
Signed-off-by: Keerthana K <keerthanak@vmware.com>

[Srinidhi Rao<srinidhi.rao@broadcom.com> Ported this patch to v6.12.x]
Signed-off-by: srinidhira0 <srinidhi.rao@broadcom.com>
---
 include/linux/cgroup.h    | 1 +
 kernel/cgroup/cgroup-v1.c | 4 ++++
 kernel/cgroup/cgroup.c    | 2 +-
 3 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/include/linux/cgroup.h b/include/linux/cgroup.h
index f8ef47f8a634..6d0a380541ed 100644
--- a/include/linux/cgroup.h
+++ b/include/linux/cgroup.h
@@ -120,6 +120,7 @@ void cgroup_file_show(struct cgroup_file *cfile, bool show);
 int cgroupstats_build(struct cgroupstats *stats, struct dentry *dentry);
 int proc_cgroup_show(struct seq_file *m, struct pid_namespace *ns,
 		     struct pid *pid, struct task_struct *tsk);
+int cgroup_stat_show(struct seq_file *seq, void *v);
 
 void cgroup_fork(struct task_struct *p);
 extern int cgroup_can_fork(struct task_struct *p,
diff --git a/kernel/cgroup/cgroup-v1.c b/kernel/cgroup/cgroup-v1.c
index e28d5f0d20ed..bd13fabf8a80 100644
--- a/kernel/cgroup/cgroup-v1.c
+++ b/kernel/cgroup/cgroup-v1.c
@@ -645,6 +645,10 @@ struct cftype cgroup1_base_files[] = {
 		.flags = CFTYPE_ONLY_ON_ROOT,
 		.seq_show = cgroup_sane_behavior_show,
 	},
+	{
+		.name = "cgroup.stat",
+		.seq_show = cgroup_stat_show,
+	},
 	{
 		.name = "tasks",
 		.seq_start = cgroup_pidlist_start,
diff --git a/kernel/cgroup/cgroup.c b/kernel/cgroup/cgroup.c
index 044c7ba1cc48..0056b45d82f5 100644
--- a/kernel/cgroup/cgroup.c
+++ b/kernel/cgroup/cgroup.c
@@ -3673,7 +3673,7 @@ static int cgroup_events_show(struct seq_file *seq, void *v)
 	return 0;
 }
 
-static int cgroup_stat_show(struct seq_file *seq, void *v)
+int cgroup_stat_show(struct seq_file *seq, void *v)
 {
 	struct cgroup *cgroup = seq_css(seq)->cgroup;
 	struct cgroup_subsys_state *css;
-- 
2.39.4

